### ISSUE TITLE
documentation question about proxy_type_t

### ISSUE THREAD

**Issue by rptaylor** [arvidn/libtorrent#7656]
http://libtorrent.org/reference-Settings.html#enum-proxy-type-t says
"The server is assumed to be an HTTP proxy. If the transport used for the connection is non-HTTP, the server is assumed to support the [CONNECT](http://tools.ietf.org/html/draft-luotonen-web-proxy-tunneling-01) method. i.e. for web seeds and HTTP trackers, a plain proxy will suffice. The proxy is assumed to not require authorization. The username and password will not be used."

Does non-HTTP transport mean TLS, i.e. HTTPS ? And what is meant by a "plain proxy"?
In other words, for a plain HTTP (not HTTPS) server (web seed), can libtorrent make a GET request through a proxy instead of using CONNECT, such that the proxy would be able to cache the content?

Thanks!

**Comment by arvidn** [comment]
> Does non-HTTP transport mean TLS, i.e. HTTPS ?

Exactly. This includes TLS trackers, TLS web seeds and regular bittorrent peer connections.

> And what is meant by a "plain proxy"?

That the `GET` request is transparent to the proxy, using the traditional HTTP proxy protocol.

> In other words, for a plain HTTP (not HTTPS) server (web seed), can libtorrent make a GET request through a proxy instead of using CONNECT, such that the proxy would be able to cache the content?

yes

**Comment by rptaylor** [comment]
Thanks!

**Comment by faarish** [comment]
"non-HTTP transport" typically refers to protocols other than HTTP, such as TCP or UDP. So, when it mentions non-HTTP transport, it doesn't specifically mean TLS (HTTPS), but rather any transport protocol other than HTTP.

**Comment by arvidn** [comment]
I agree, this part of the documentation could be improved.

**Comment by rptaylor** [comment]
@faarish But HTTP and TCP/UDP are different layers (layer 7 and layer 4, respectively).  HTTP is not a transport protocol in the first place, and any HTTP connection must by necessity also use a transport layer protocol such as TCP or UDP (e.g. QUIC for HTTP/3). Moreover, CONNECT is only for tunnelling TCP, not UDP.

Instead of "If the transport used for the connection is non-HTTP, the server is assumed to support the [CONNECT](http://tools.ietf.org/html/draft-luotonen-web-proxy-tunneling-01) method" should it say something like "If the transport layer protocol used for the connection is TCP, and either TLS is used or the application layer protocol is not HTTP, the server is assumed to support the [CONNECT](http://tools.ietf.org/html/draft-luotonen-web-proxy-tunneling-01) method"  ?
